src/main/scala/org/apache/spark/sql/catalyst/plans/logical/merge.scala

```diff
@@ -244,7 +244,7 @@ object MergeInto {
       resolvedExpr.flatMap(_.references).filter(!_.resolved).foreach { a =>
         // Note: This will throw error only on unresolved attribute issues,
         // not other resolution errors like mismatched data types.
-        val cols = "columns " + plan.references.map(_.sql).mkString(", ")
```

Comment:
`references` includes refs to invalid column names as well, which we do not want to print as valid names.